### ISSUE TITLE
Panasonic TV only recognized via UPnP if server started after TV

### ISSUE THREAD

**Issue by SubJunk** [UniversalMediaServer/UniversalMediaServer#918]
This issue has been around for ages and we gave up on it around version 5. I'm motivated to work on it again, especially since I think the same thing might improve our connectivity with other devices.

If the server is started after the TV, the TV is recognized by its specific name "VIERA VT60", but if the TV is started after the server, it only matches the generic "Panasonic TV" renderer profile.

This means it's something we can fix - the TV is reacting to something we are sending on startup, but aren't sending later.

This also happens for Panasonic ST50 TVs so I suspect it affects many Panasonic TVs.

I can't remember where we got to with this last time, do you remember what happened with it @skeptical ?

**Comment by Sami32** [comment]
@SubJunk I remember have this kind of problem, i "solved" it by using (it give me mainly the default Viera icon from my renderer until i use _Refresh_)
```
UserAgentAdditionalHeader = X-PANASONIC-Registration
UserAgentAdditionalHeaderSearch = VklFUkEgRTYgU2VyaWVzAAAAAAAAAAAAAAA=
```
We can see it appear on the renderer before rapidly disappear for maybe 30s before appear for good this time.

I get this problem if Firefox is used at the same time as well, so UMS will not be detected by my Panasonic Viera TV.

**Comment by Sami32** [comment]
If i run UMS **after** put on the TV _Wireshark_ give me :

> GET /dmr/ddd.xml HTTP/1.1
> User-agent: UMS/6.3.3-SNAPSHOT WindowsXP/5.1 UPnP/1.0 Cling/2.0
> Host: 192.168.1.69:55000
> Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
> Connection: close
> 

> GET /dmr/sdd_0.xml HTTP/1.1
> GET /dmr/sdd_1.xml HTTP/1.1
> GET /dmr/sdd_2.xml HTTP/1.1
> GET /dmr/dlna_icon_120.png HTTP/1.1
> SUBSCRIBE /dmr/event_0 HTTP/1.1
> SUBSCRIBE /dmr/event_1 HTTP/1.1
> SUBSCRIBE /dmr/event_2 HTTP/1.1
> POST /dmr/control_2 HTTP/1.1

If i run UMS **before** open the TV i get

> GET /description/fetch HTTP/1.1
> User-Agent: Panasonic MIL DLNA CP UPnP/1.0 DLNADOC/1.50
> Host: 192.168.1.28:5001
> X-PANASONIC-Registration: VklFUkEgRTYgU2VyaWVzAAAAAAAAAAAAAAA=
> X-PANASONIC-DMP-Profile: MPEG_PS_PAL JPEG_SM PV_DIVX_DIV3 PV_DIVX_DIV4 PV_DIVX_DIVX PV_DIVX_DX50 PV_DRM_DIVX_DIV3 PV_DRM_DIVX_DIV4 PV_DRM_DIVX_DIVX PV_DRM_DIVX_DX50
> 
> GET /images/icon-256.png HTTP/1.1
> User-Agent: Panasonic MIL DLNA CP UPnP/1.0 DLNADOC/1.50
> Host: 192.168.1.28:5001
> transferMode.dlna.org: Interactive

> POST /upnp/control/content_directory HTTP/1.1

**Comment by SubJunk** [comment]
Nice work! I'm looking into this now too

**Comment by SubJunk** [comment]
So far I can't see what's different. Maybe I'm not great with Wireshark but I can't see a difference between what we send when UMS starts and what we send regularly when it's running. I'll keep looking. Does anyone have an idea @UniversalMediaServer/developers ?

**Comment by SubJunk** [comment]
Okay I spent the night playing around and have a possible fix at https://github.com/UniversalMediaServer/UniversalMediaServer/pull/920
Please test and let me know

**Comment by SubJunk** [comment]
The fix for this was released in 6.4.0
